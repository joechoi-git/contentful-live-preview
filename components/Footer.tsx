import React from "react";
import Link from "next/link";

export default function Footer() {
    return (
        <footer className="w-full text-center border-t border-slate-300 pt-12">
            <p>This is a sample app built to demonstrate the Contentful Live Preview feature.</p>
            <p>
                <Link
                    href="https://github.com/joechoi-git/contentful-live-preview"
                    target="_blank"
                    className="text-blue-500"
                >
                    Open GitHub repository.
                </Link>
            </p>
            <p className="pt-6">
                &copy; {new Date().getFullYear()}. All rights reserved by Joe Choi.
            </p>
        </footer>
    );
}
