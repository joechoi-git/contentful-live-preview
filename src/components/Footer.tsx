import React from "react";
import Link from "next/link";

export default function Footer() {
    return (
        <footer className="mt-6 w-full text-center border-t-2 border-secondary py-6">
            <p>This is a sample app built to demonstrate the Contentful Live Preview feature.</p>
            <p>
                <Link
                    href="https://github.com/joechoi-git/contentful-live-preview"
                    target="_blank"
                    className="text-primary"
                >
                    Open GitHub repository.
                </Link>
            </p>
            <p className="pt-6">&copy; {new Date().getFullYear()}</p>
        </footer>
    );
}
