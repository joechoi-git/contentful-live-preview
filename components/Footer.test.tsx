/* eslint-disable indent */
import React from "react";
import { render } from "@testing-library/react";
import Footer from "../components/Footer";

jest.mock("../locales/client", () => ({
    useI18n: jest.fn()
}));

describe("Footer", () => {
    it("renders correctly with i18n translations", () => {
        const mockI18n = jest.requireMock("../locales/client").useI18n;
        mockI18n.mockReturnValue((key: any) => {
            switch (key) {
                case "sampleApp":
                    return "Sample Application";
                case "openGitHub":
                    return "Open GitHub";
                default:
                    return key;
            }
        });

        const { getByText, asFragment } = render(<Footer />);

        expect(getByText("Sample Application")).toBeInTheDocument();
        expect(getByText("Open GitHub")).toBeInTheDocument();
        expect(getByText(`Â© ${new Date().getFullYear()}`)).toBeInTheDocument();

        expect(asFragment()).toMatchSnapshot();
    });
});
