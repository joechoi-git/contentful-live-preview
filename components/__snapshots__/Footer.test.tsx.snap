// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly with i18n translations 1`] = `
<DocumentFragment>
  <footer
    class="mt-6 w-full text-center border-t-2 border-secondary py-6"
  >
    <p>
      Sample Application
    </p>
    <p>
      <a
        class="text-accent"
        href="https://github.com/joechoi-git/contentful-live-preview"
        target="_blank"
      >
        Open GitHub
      </a>
    </p>
    <p
      class="pt-6"
    >
      Â© 2024
    </p>
  </footer>
</DocumentFragment>
`;
